internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -365,7 +366,7 @@ func resourcePostgresqlFlexibleServerCreate(d *pluginsdk.ResourceData, meta inte
 	updateProperties := servers.ServerPropertiesForUpdate{}
 	if authRaw, ok := d.GetOk("authentication"); ok {
 		authConfig := expandFlexibleServerAuthConfig(authRaw.([]interface{}))
-		if !*authConfig.PasswordAuthEnabled {
+		if *authConfig.PasswordAuth == servers.PasswordAuthEnumDisabled {
```

Comment:
Since we're changing from a bool to a string comparison, I think we'll need to start nil-checking this here:
```suggestion
		if authConfig != nil && authConfig.PasswordAuth != nil && *authConfig.PasswordAuth == servers.PasswordAuthEnumDisabled {
```

Reply:
Updated